core/esmf-aspect-meta-model-java/src/main/java/org/eclipse/esmf/aspectmodel/loader/Instantiator.java

```diff
@@ -116,7 +116,14 @@ protected Type getType( final Resource characteristicResource ) {
    private Statement getDataType( final Resource resource ) {
       return Optional.ofNullable( resource.getPropertyResourceValue( SammNs.SAMMC.baseCharacteristic() ) )
             .map( this::getDataType )
-            .orElseGet( () -> resource.getProperty( SammNs.SAMM.dataType() ) );
+            .orElseGet( () -> {
+               final Statement dataType = resource.getProperty( SammNs.SAMM.dataType() );
+               if ( dataType == null ) {
+                  throw new IllegalStateException(
```

Comment:
The only exception that should be thrown by the Instantiators is `AspectLoadingException`.